### PR TITLE
Update sandboxes to use vite and typescript

### PR DESCRIPTION
**Pull request by ChristianMurphy**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

the native dependencies in parcel no longer work in browser sandboxes like stackblitz
create react app has been (for all intents and purposes) deprecated by the react team
vite offers a replacement for both

also run sandboxes through typescript to catch potential issues with plugin compatibility and options.

temporary stackblitz links to test with:
https://stackblitz.com/github/ChristianMurphy/.github-remark/tree/migrate-sandboxes-to-vite/sandbox-templates/remark-with-vite
https://stackblitz.com/github/ChristianMurphy/.github-remark/tree/migrate-sandboxes-to-vite/sandbox-templates/remark-rehype-with-vite
https://stackblitz.com/github/ChristianMurphy/.github-remark/tree/migrate-sandboxes-to-vite/sandbox-templates/react-markdown-with-vite

CodeSandbox currently has issues with any repo with a `.` in it's name getting blocked by cloudflare or triggering a not authorized error.

### CODE REVIEW

File: sandbox-templates/remark-rehype-with-vite/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "lib": ["ES2020", "DOM", "DOM.Iterable"],
+    "module": "Node16",
+    "isolatedModules": true,
+    "noEmit": true,
+    "jsx": "react-jsx",
+    "strict": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noFallthroughCasesInSwitch": true,
+    "skipLibCheck": true
```
Comment (ChristianMurphy) [inline sandbox-templates/remark-rehype-with-vite/tsconfig.json:14]: this should be removed, currently it is blocked by https://github.com/rehypejs/rehype/issues/147

File: sandbox-templates/react-markdown-with-vite/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "react-markdown-with-vite",
+  "version": "1.0.0",
+  "private": true,
+  "main": "src/main.tsx",
+  "type": "module",
+  "dependencies": {
+    "react": "^18.0.0",
```
Comment (ChristianMurphy) [inline sandbox-templates/react-markdown-with-vite/package.json:8]: this also updates to React 18

File: package.json [hunk 3]
```diff
@@ -48,7 +48,8 @@
           "sandbox-templates/**/*.tsx"
         ],
         "rules": {
-          "@typescript-eslint/naming-convention": "off"
+          "@typescript-eslint/naming-convention": "off",
+          "react/react-in-jsx-scope": "off"
```
Comment (ChristianMurphy) [inline package.json:52]: the vite react transform does not require react in scope

File: sandbox-templates/react-markdown-with-vite/src/main.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+// eslint-disable-next-line n/file-extension-in-import
+import ReactDOM from 'react-dom/client'
```
Comment (ChristianMurphy) [inline sandbox-templates/react-markdown-with-vite/src/main.tsx:3]: this lint rule will break the code by adding a `.js` extension.
`react-dom/client` is in the export map, `react-dom/client.js` is not a thing.

**Review by wooorm — APPROVED** [review]

**Pull request merged**